lib/internal/test_runner/test.js

```diff
@@ -998,7 +1000,18 @@ class Test extends AsyncResource {
       process.exit();
     }
   }
-
+  getFiles(files = []) {
```

Comment:
Is there a way to get/store the files from where they are actually gotten from, to remove all of this logic?

Reply:
I think I'm going to do the "store" approach. Was my first try, but then I went to this more functional one, but it ended modifying tons of files just to send the argument from one function to the other...

Reply:
Well, couldn't it just be added to the test options when it is gotten? All the options from the test are passed to the coverage reporter.

Reply:
The problem with that is the watcher, which updates the test files on the fly, so we need a way to update this list on the fly too.

Reply:
Is that true? I don't think it updates _which_ files are run

Reply:
Yes. If you search for the `watcher.on('changed'` event inside `lib/internal/test_runner/runner.js` you'll see that there's a new `createTestFileList` each time something changes.